File: Pila_Arreglo/Pila.js
const readline = require("readline");
class Stack {
    constructor() {
        this.MAX_SIZE = 100;
        this.stack = [];
    }
    push(item) {
        if (this.stack.length === this.MAX_SIZE) return console.log("Stack Overflow");
        this.stack.push(item);
    }
    pop() {
        if (this.stack.length === 0) return console.log("Stack Underflow");
        return this.stack.pop();
    }
    peek() {
        if (this.stack.length === 0) return console.log("Pila vacía");
        return this.stack[this.stack.length - 1];
    }
    isEmpty() { return this.stack.length === 0; }
    isFull() { return this.stack.length === this.MAX_SIZE; }
}

const pila = new Stack();
const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

function menu() {
    console.log("\n--- MENU PILA (JS) ---");
    console.log("1. Push");
    console.log("2. Pop");
    console.log("3. Peek");
    console.log("4. isEmpty");
    console.log("5. isFull");
    console.log("6. Salir");
    rl.question("Elige: ", opcion => {
        switch(parseInt(opcion)) {
            case 1:
                rl.question("Ingresa un valor: ", v => { pila.push(parseInt(v)); menu(); });
                break;
            case 2:
                console.log("Eliminado:", pila.pop()); menu(); break;
            case 3:
                console.log("Tope:", pila.peek()); menu(); break;
            case 4:
                console.log(pila.isEmpty()? "Pila vacía" : "No está vacía"); menu(); break;
            case 5:
                console.log(pila.isFull()? "Pila llena" : "No está llena"); menu(); break;
            case 6:
                console.log("Saliendo..."); rl.close(); break;
            default:
                console.log("Opción inválida"); menu(); break;
        }
    });
}
menu();

File: Pila_Arreglo/Pila.cs
using System;

class StackMenu {
    const int MAX_SIZE = 100;
    int[] stack = new int[MAX_SIZE];
    int top = -1;

    void Push(int item) {
        if (top == MAX_SIZE - 1) { Console.WriteLine("Stack Overflow"); return; }
        stack[++top] = item;
    }

    int Pop() {
        if (top == -1) { Console.WriteLine("Stack Underflow"); return -1; }
        return stack[top--];
    }

    int Peek() {
        if (top == -1) { Console.WriteLine("Pila vacía"); return -1; }
        return stack[top];
    }

    bool IsEmpty() => top == -1;
    bool IsFull() => top == MAX_SIZE - 1;

    static void Main() {
        StackMenu pila = new StackMenu();
        int opcion, valor;

        do {
            Console.WriteLine("\n--- MENU PILA (C#) ---");
            Console.WriteLine("1. Push");
            Console.WriteLine("2. Pop");
            Console.WriteLine("3. Peek");
            Console.WriteLine("4. isEmpty");
            Console.WriteLine("5. isFull");
            Console.WriteLine("6. Salir");
            Console.Write("Elige: ");
            opcion = int.Parse(Console.ReadLine());

            switch(opcion) {
                case 1:
                    Console.Write("Ingresa un valor: ");
                    valor = int.Parse(Console.ReadLine());
                    pila.Push(valor);
                    break;
                case 2:
                    valor = pila.Pop();
                    if(valor != -1) Console.WriteLine("Eliminado: " + valor);
                    break;
                case 3:
                    valor = pila.Peek();
                    if(valor != -1) Console.WriteLine("Tope: " + valor);
                    break;
                case 4:
                    Console.WriteLine(pila.IsEmpty()? "Pila vacía" : "No está vacía");
                    break;
                case 5:
                    Console.WriteLine(pila.IsFull()? "Pila llena" : "No está llena");
                    break;
                case 6:
                    Console.WriteLine("Saliendo...");
                    break;
                default:
                    Console.WriteLine("Opción inválida");
                    break;
            }
        } while(opcion != 6);
    }
}
